src/clinical/service-worker-thread/tasks.ts

```diff
@@ -190,18 +190,31 @@ const mapEntityDocuments = (
 // Submitted Data Search Results
 function FilterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
   const { donorIds, submitterDonorIds } = query;
+  const useQueriedDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
 
-  const useFilteredDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
-
-  const filteredDonors = useFilteredDonors
-    ? donors.filter(donor => {
+  const filteredDonors = donors
+    .filter(donor => {
+      if (useQueriedDonors) {
         const { donorId, submitterId } = donor;
         const stringId = `${donorId}`;
         const donorMatch = donorIds?.some(id => stringId.includes(id));
         const submitterMatch = submitterDonorIds?.some(id => submitterId.includes(id));
         return donorMatch || submitterMatch;
-      })
-    : donors;
+      }
+      return donor;
+    })
+    .filter(donor => {
+      // This filters out false positive search results ( i.e. where Donor.treatments = [] )
+      const queriedEntities = Object.values(ClinicalEntitySchemaNames).filter(entity =>
+        isEntityInQuery(entity, query.entityTypes),
+      );
+
+      const clinicalInfoRecords = queriedEntities
+        .map(entity => getClinicalEntitySubmittedData(donor, entity))
+        .filter(record => !isEmpty(record));
```

Comment:
i think `getClinicalEntitySubmittedData` always returns a base record?

Reply:
Yes you are correct, there's a deeper issue here where Sample Registrations are tied to Specimens which creates empty Specimen records, which I need to untangle. I will revisit this and post update.

Reply:
OK I gave this a deeper look and need to confirm desired behaviour for filtering empty clinical Info, will move all filtering into getClinicalEntitySubmittedData

Reply:
Revised this today. `getClinicalEntitySubmittedData` always returns a map of the data returned from `getClinicalObjectsFromDonor`. If that data is an empty array, there is nothing to map, so an empty array is returned. I changed the default value of `clinicalRecords` to `[]` i.e. removed the base record.

Comments are updated and record counts are accurate on localhost

This function is solving the problem where `Donor[clinicalEntity] = []`
These were being counted as 'searchResults' of the relevant clinicalEntity
This is different from `Donor[clinicalEntity] = [ { } ]`, where data has been submitted but is incomplete, which is where the baseRecord comes into play